### DISCUSSION TITLE
9월 26일 주간회의

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#1041] [General]
현재 진행중인 작업에 대해서 공유할 내용이 있으면 코멘트로 공유해주세요~

**Comment by wishoon** [comment]
### 레벨로그 댓글 
* PR Review 진행 #1033

**Comment by al-bur** [comment]
### 내 학습로그 뒤로 가기 버그
- dev 배포후 테스트시 필터 기능중 일부분에 버그 발생하여 revert => 임시저장 기능 구현 후 작업 예정

### 학습로그 임시 저장 https://github.com/woowacourse/prolog/issues/765
- 작업 진행중 (~10월 6일까지)
  - 임시 저장 버튼 UI 구현
  - 임시 저장 조회 api, type 로직 구현
  - 현재, 에디터 컴포넌트에 대해서 어떻게 구현해야할 지 고민중

**Comment by verus-j** [comment]
### 로컬에서 flyway가 정상 동작하는지 확인할 수 있도록 docker를 이용해 세팅
* PR 생성 #1042

**Comment by hanull** [comment]
### 이미지 삽입 기능
- https://github.com/woowacourse/prolog/issues/62

### 진행 상황
- [x] S3 버킷, CloudFront 생성
- [x] 이미지 파일 이름이 빈 경우 예외
- [x] 이미지 파일 확장자가 잘못된 경우 예외
- [ ] service 테스트 작성(LocalStack 사용) 
- [ ] Documentation 작성

**Reply by Yboyu0u** [reply]
Documentation은 NewDocumentation 사용하면 좋겠네요👍🏻

**Comment by kwannee** [comment]
### 피그마 정리
- 디자인 시스템 기본적인 단위 정리 ( 진행 중 )

### 프론트엔드 개발 방법 문서화
- 프론트엔드 서버 구동 오류 시 방법 문서화
- 백엔드 서버 구동 방법 문서화

**Comment by JUDONGHYEOK** [comment]
현재 이미지 관련작업 이슈 발생해서 대기중이고 comment작업 UI완료했고 API관련 브라운과 협의 예정입니다.:)

**Comment by leo0842** [comment]
#1027 

### 모니터링 문서화
- 현재 상태와 한계 이유 수집

### TODO
- 각각의 기술(promtail, loki, grafana) 사용 이유 
- 위의 사항들을 정리하여 문서화

**Comment by euijinkk** [comment]
## 로드맵 기획 및 UI 작업

![로드맵](https://user-images.githubusercontent.com/24906022/192204470-a47cf74c-8f2f-4c45-b28f-b83f996f8989.jpg)

**Comment by Yboyu0u** [comment]
## 테스트개선 - 문서화 테스트 
###  Document(RestAssured+ @SpringbootTest)사용하는 테스트를 -> NewDocument(Mockmvc)로 바꾸는 작업 
ex)
```java
    @Test
    void 사용자_프로필을_조회한다() {
        //given
        given(memberService.findMemberResponseByUsername(any()))
                .willReturn(MEMBER_RESPONSE);

        //when
        ValidatableMockMvcResponse response = given
                .accept(MediaType.APPLICATION_JSON_VALUE)
                .when().get("/members/{username}/profile", GithubResponses.소롱.getLogin())
                .then().log().all();

        //then
        response.expect(status().isOk());

        //docs
        response.apply(document("profile/profile"));
    }
```

- Document 사용 문서화 테스트 총 10개 중 3개 완료 
      ~~Login~~
~~Mission~~
~~Profile~~
~~Comment~~
~~Member~~
~~MemberReaction~~
~~SessionMember~~
~~Tag~~
Studylog
PopularStudylog
StudylogLike
~~StudylogoverView~~
StudylogRssFeed

- 문서화를 위해 스터빙될 Response들을  fixture로 빼내는 작업 진행 
- 의미있는 문서를 만들기 위해 response를 만드는 과정에서 시간이 많이 소요됨 
ex)
```java
   public static final MissionResponse MISSION_RESPONSE1 =
            new MissionResponse(1L, "지하철 노선도 미션", SESSION_RESPONSE);

    public static final MemberResponse MEMBER_RESPONSE = new MemberResponse(
            1L,
            "soulG",
            "잉",
            CREW,
            "https://avatars.githubusercontent.com/u/52682603?v=4"
    );

    private static final StudylogResponse STUDY_LOG_RESPONSE1 = new StudylogResponse(
            1L,
            MEMBER_RESPONSE,
            LocalDateTime.now(),
            LocalDateTime.now(),
            SESSION_RESPONSE,
            MISSION_RESPONSE1,
            "JAVA",
            "Spring Data JPA를 학습함.",
            List.of(new TagResponse(3L, "java"), new TagResponse(4L, "jpa")),
            List.of(),
            false,
            false,
            0,
            false,
            0
    );

```
- 할당 이슈: #1004
- 작업중인 브랜치: [브랜치](https://github.com/woowacourse/prolog/tree/refactor/1004-DocumentationToNewDocumentation)
